File: Easy/Coin Flip (CONFLIP)/coinflip.cpp
#include <iostream>
using namespace std;

int main() {
	// your code goes here
	int testCase;
	cin>>testCase;
	for(int y = 0;y<testCase;y++)
	{
    	int noOfRounds;
    	cin>>noOfRounds;
    	for(int x =0;x < noOfRounds;x++)
    	{
        	long int i,n,q;
        	cin>>i >> n >> q;
            if(i == 1)        	
            {
                if(q==1)
                {
                    cout<< n/2<<endl;
                    
                }
                else
                {
                    cout << n - n/2<<endl;
                }
            }
            else
            {
                if(q==2)
                {
                    cout<< n/2<<endl;
                    
                }
                else
                {
                    cout << n - n/2<<endl;
                }                
            }
    	}
	}
    
	
	return 0;
}

File: Easy/Coin Flip (CONFLIP)/coinflip.js
process.stdin.resume();
process.stdin.setEncoding("utf8");

// your code goes here
let input = "";
process.stdin.on("data", (chunk) => (input += chunk));
process.stdin.on("end", () => {
  input = input.trim().split("\n");
  let t = input[0];
  let i = 1;
  let G;
  let N;
  let j;
  while (t--) {
    G = input[i++];
    while (G--) {
      j = input[i++].split(" ");
      N = j[1];
      if (N % 2 === 0) {
        process.stdout.write(N / 2 + "\n");
      } else {
        process.stdout.write((j[0] == j[2] ? N - 1 : +N + 1) / 2 + "\n");
      }
    }
  }
});
